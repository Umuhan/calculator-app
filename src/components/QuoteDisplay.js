import React, { useEffect, useState } from 'react';

const QuoteDisplay = () => {
  const [quote, setQuote] = useState({
    loading: false,
    error: false,
    loadingData: [],
  });
  useEffect(() => {
    const fetchQuotes = async () => {
      try {
        setQuote((State) => ({ ...State, loading: true }));
        const res = await fetch(
          'https://api.api-ninjas.com/v1/quotes?category=family',
          {
            headers: {
              'X-Api-Key': 'iBbkx3sRtVSUUTVrcH90+A==li5LiENpC7ZgNO5z',
            },
          },
        );

        if (!res.ok) {
          throw new Error('Error Occurred...');
        }
        const result = await res.json();

        setQuote((State) => ({
          ...State,
          loadingData: result,
          loading: false,
        }));
      } catch (err) {
        setQuote((State) => ({ ...State, error: true, loading: false }));
      }
    };
    fetchQuotes();
  }, []);
  let content = '';
  if (quote.loading) content = 'Loading ...';
  if (quote.error) content = 'Error Occurred...';

  return (
    <>
      {quote.error ? (
        <p className="error">{content}</p>
      ) : (
        <div className="Container">
          <h2 className="Header">Quotes</h2>
          <p className="loading">{content}</p>
          {quote.loadingData.map((data) => (
            <div className="Text" key={data.id}>
              <h2>{`"${data.category}":`}</h2>
              <p>{`"${data.quote}" - ${data.author}`}</p>
            </div>
          ))}
        </div>
      )}
    </>
  );
};
export default QuoteDisplay;
