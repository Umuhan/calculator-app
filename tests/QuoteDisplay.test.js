import React from "react";
import renderer from "react-test-renderer";
import QuoteDisplay from "../src/components/QuoteDisplay";

it("renders correctly", () => {
  const tree = renderer.create(<QuoteDisplay />).toJSON();
  expect(tree).toMatchSnapshot();
});
