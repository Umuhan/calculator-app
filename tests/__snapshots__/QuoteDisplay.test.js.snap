// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <h1
    className="heading"
    style={
      Object {
        "textAlign": "left",
      }
    }
  >
    Math Magicians
  </h1>
  <div
    className="Container"
  >
    <p
      className="loading"
    />
  </div>
</div>
`;
